src/routes/DigitalAssets/routes/Send/sagas/digitalAssetsSend.js

```diff
@@ -516,27 +546,89 @@ function* setPriority(): Saga<void> {
     const digitalAsset: ExtractReturn<typeof selectDigitalAsset> =
       yield select(selectDigitalAsset, assetAddress)
 
-    if (!((digitalAsset &&
-          digitalAsset.blockchainParams &&
-          digitalAsset.blockchainParams.staticGasAmount)
-          || digitalAsset.isCustom)) {
-      // we can't call estimateGas for custom asset here, because we need a privateKey for it
+    if (!digitalAsset) {
       return
     }
 
-    yield put(
-      digitalAssetsSend.setFormFieldValue('gasLimit', digitalAsset.blockchainParams.staticGasAmount)
-    )
+    const gasLimit = yield* requestGasLimit(digitalAsset)
+    if (gasLimit) {
+      yield put(digitalAssetsSend.setFormFieldValue('gasLimit', String(gasLimit)))
+    }
+  }
+}
+
+function* checkGasLimitWarning(newGasLimit: string): Saga<void> {
+  const {
+    formFieldWarnings,
+    initialGasSettings,
+  }: DigitalAssetsSendState = yield select(selectDigitalAssetsSend)
+
+  // Priority warnings
+  const gasLimit = parseInt(newGasLimit, 10)
+  const initialGasLimit = parseInt(initialGasSettings.gasLimit, 10)
+
+  if (!Number.isNaN(gasLimit) &&
+      !Number.isNaN(initialGasLimit) &&
+      gasLimit < initialGasLimit) {
+    yield put(digitalAssetsSend.setFormFieldWarning(
+      'gasLimit',
+      'Gas limit is too small. Most likely, your transaction will fail'
+    ))
+  } else if (formFieldWarnings.gasLimit) {
+    yield put(digitalAssetsSend.setFormFieldWarning('gasLimit', ''))
+  }
+}
+
+function* checkGasPriceWarning(newGasPrice: void | string | BigNumber): Saga<void> {
+  const {
+    formFieldWarnings,
+    initialGasSettings,
+  }: DigitalAssetsSendState = yield select(selectDigitalAssetsSend)
+
+  // Priority warnings
+  try {
+    const gasPrice = toBigNumber(newGasPrice)
+    const initialGasPrice = toBigNumber(initialGasSettings.gasPrice)
+
+    if (gasPrice.lt(initialGasPrice.times(MIN_GAS_PRICE_COEFFICIENT)) ||
+      gasPrice.lt(ETH_MIN_GAS_PRICE)
+    ) {
+      yield put(digitalAssetsSend.setFormFieldWarning(
+        'gasPrice',
+        'Gas price is too small. Most likely, your transaction will fail'
+      ))
+    } else if (formFieldWarnings.gasPrice) {
+      yield put(digitalAssetsSend.setFormFieldWarning('gasPrice', ''))
+    }
+  } catch (err) {
+    // when toBigNumber throws exception - user input is invald, it is error, not warning
+    if (formFieldWarnings.gasPrice) {
```

Comment:
we have to prevent situation when `toBigNumber` trows exception
please validate arguments instead of writing such dirty code

otherwise everyone must know that this function will generate errors in different situations via `console.error` into this `catch` block.